### PR TITLE
Return python lists for __getitem__ calls to list type series

### PR DESCRIPTION
**Pull request by brandon-b-miller**
Make it so that this works:

```
x = cudf.Series([[1,2,None]])
x[0]
# [1, 2, <NA>]
```

### CODE REVIEW

File: python/cudf/cudf/tests/test_list.py [hunk 1]
```diff
@@ -332,3 +332,20 @@ def test_concatenate_list_with_nonlist():
...
+        [[1, 2, 3], [4, 5, 6]],
+        [None],
+        [1, None, 3],
+        [[1, None, 3], [None, 5, 6]],
+    ],
+)
+def test_list_getitem(data):
+    list_sr = cudf.Series([data])
+    # __getitem__ shall fill None with cudf.NA
+    assert pa.scalar(data) == list_sr.to_arrow()[0]
```
Comment (isVoid) [inline python/cudf/cudf/tests/test_list.py:351]: Is this actually testing our `__getitem__` code path?

Comment (brandon-b-miller) [reply python/cudf/cudf/tests/test_list.py]: oops :) fixed.

**Review by galipremsagar — APPROVED** [review]

**Comment by galipremsagar** [comment]
rerun tests

**Review by isVoid — APPROVED** [review]

**Comment by brandon-b-miller** [comment]
@gpucibot merge

**Pull request merged**